mga/update-network-node.go

```diff
@@ -32,6 +33,16 @@ func UpdateNetworkNodeFx(w http.ResponseWriter, r *http.Request) (string, interf
 		return "", "", err
 	}
 
+	_, err = network.GetNetworkNodeByCode(inputNode.Code)
+	if !errors.Is(err, network.ErrNodeNotFound) {
+		// If err is *not* ErrNodeNotFound, options are:
+		// - an error raised hitting the db
+		// - the node code is already present
+		// In both cases we must abort
+		log.Printf("error checking node code: %s", err.Error())
+		return "", "", err
+	}
```

Comment:
Even though a `nil` error gets catched by the control flow, we would have a panic here for the log line since `nil` does not have an `Error()` method. Regardless, the return response does not suffice in this case because we would return a `statusCode 200` with an empty body to the FE. In this case we must ensure that we return _some_ kind of error.

It seems we need to split the flow:
- error `nil` => node found so we exit with another type of error
- error not `nil` && not `ErrNodeNotFound` => we got some kind of unexpected error so we exit with that error